src/tools/index.py

```diff
@@ -86,3 +88,107 @@ def get_index_advisor_recommendations(
     except Exception as e:
         logger.error(f"Error running Index Advisor: {e!s}", exc_info=True)
         raise
+
+
+def list_indexes(
+    ctx: Context,
+    bucket_name: str | None = None,
+    scope_name: str | None = None,
+    collection_name: str | None = None,
+    include_raw_index_stats: bool = False,
+) -> list[dict[str, Any]]:
+    """List all indexes in the cluster with optional filtering by bucket, scope, and collection.
+    Returns a list of indexes with their names and CREATE INDEX definitions.
+    Uses the Index Service REST API (/getIndexStatus) to retrieve index information directly.
+
+    Args:
+        ctx: MCP context for cluster connection
+        bucket_name: Optional bucket name to filter indexes
+        scope_name: Optional scope name to filter indexes (requires bucket_name)
+        collection_name: Optional collection name to filter indexes (requires bucket_name and scope_name)
+        include_raw_index_stats: If True, include raw index stats (as-is from API) in addition
+                              to cleaned-up version. Default is False.
+
+    Returns:
+        List of dictionaries with keys:
+        - name (str): Index name
+        - definition (str): Cleaned-up CREATE INDEX statement
```

Comment:
Can we add the following fields to the normal response?
`status`, `isPrimary`, `bucket`, `scope`, `collection`